### ISSUE TITLE
Text Editor has no required attribute for screen readers

### ISSUE THREAD

**Issue by samtjo** [Sage/carbon#3326]
### Current behaviour
The Text Editor input has no `required` or `aria-required` attributes so it doesn't get picked up as required by a screen reader.

### Expected behaviour
The screen reader should be able to recognise the editor field is required.

### Reproducible example
See this story: https://carbon.sage.com/?path=/story/design-system-text-editor--with-content 

- Inspect the label and content
- Turn on the VoiceOver screenreader (CMD + F5)
- Select the label "Text Editor Label"

The label indicates it's required but it does not get picked up by a screen reader as so.

### Suggested solution(s)
It isn't possible to add the `required` attribute as there are no input elements in the text editor but it should be possible to add `aria-required`. 

Unfortunately the `draft-js` editor being used does not support that currently so a PR adding that should be raised.

### Your environment
| Software         | Version(s) |
| ---------------- | ---------- |
| carbon-react     | Latest
| carbon-factory   |
| react-scripts    | 
| React            |
| Browser          |
| npm              |
| Operating System |

**Comment by sjohnson-sage** [comment]
PR raised in draft-js to add the `aria-required` attribute here: https://github.com/facebook/draft-js/pull/2675

This issue can't move forward until this is merged.

**Comment by sjohnson-sage** [comment]
FE-3209

**Comment by jamime** [comment]
Checked PR, no progress yet.

**Comment by nicktitchmarsh** [comment]
PR still open

**Comment by jamime** [comment]
Third party PR has been merged. Removing `triage`, FE ticket will progress as normal.

**Comment by ianforster-sage** [comment]
Although the PR has been merged there hasn't been a release since 17 Aug 2020. Adding `triage` label once again.
https://github.com/facebook/draft-js/releases

**Comment by dawidzarzycki** [comment]
We are still waiting for release 
https://github.com/facebook/draft-js/releases

**Comment by nicktitchmarsh** [comment]
Still waiting for draft-js to be released

**Comment by DipperTheDan** [comment]
Still waiting for draft-js to release.

**Comment by dawidzarzycki** [comment]
Still waiting for draft-js to release.

**Comment by jamime** [comment]
https://github.com/facebook/draft-js/releases/tag/v0.11.7 is still the most recent release.

**Comment by sjohnson-sage** [comment]
I've asked on my draft-js PR when the next version will be released, will update if I get a response

**Comment by mmaciorowski-sage** [comment]
@sjohnson-sage  did you get any response?

**Comment by jamime** [comment]
https://github.com/facebook/draft-js/releases
https://github.com/facebook/draft-js/releases/tag/v0.11.7 is still the most recent release. Previous maintainer has said that they don't work at Facebook and they are unsure who to ask to cut a release 😱

**Comment by jamime** [comment]
https://github.com/facebook/draft-js/releases/tag/v0.11.7 is still the most recent release.

**Comment by nicktitchmarsh** [comment]
I think we need to draw a line under this, when Design System require and update of the Text Editor we should consider refactoring and using a better supported library.

**Comment by Parsium** [comment]
Passed on roadmap to find a replacement for `draft.js` so will close this
